src/parcsr_mv/par_csr_matrix.h

```diff
@@ -127,16 +127,31 @@ typedef struct hypre_ParCSRMatrix_struct
 static inline HYPRE_MemoryLocation
 hypre_ParCSRMatrixMemoryLocation(hypre_ParCSRMatrix *matrix)
 {
-   HYPRE_MemoryLocation memory_diag = hypre_CSRMatrixMemoryLocation(hypre_ParCSRMatrixDiag(matrix));
-   HYPRE_MemoryLocation memory_offd = hypre_CSRMatrixMemoryLocation(hypre_ParCSRMatrixOffd(matrix));
+   if (!matrix) { return HYPRE_MEMORY_UNDEFINED; }
 
-   if (memory_diag != memory_offd)
+   hypre_CSRMatrix *diag = hypre_ParCSRMatrixDiag(matrix);
+   hypre_CSRMatrix *offd = hypre_ParCSRMatrixOffd(matrix);
+   HYPRE_MemoryLocation memory_diag = diag ? hypre_CSRMatrixMemoryLocation(diag) : HYPRE_MEMORY_UNDEFINED;
+   HYPRE_MemoryLocation memory_offd = offd ? hypre_CSRMatrixMemoryLocation(offd) : HYPRE_MEMORY_UNDEFINED;
+
+   if (diag && offd)
    {
-      hypre_printf("Warning: ParCSRMatrix Memory Location Diag (%d) != Offd (%d)\n", memory_diag,
-                   memory_offd);
-      hypre_assert(0);
+      if (memory_diag != memory_offd)
+      {
+         hypre_printf("Error: ParCSRMatrix Memory Location Diag (%d) != Offd (%d)\n", memory_diag, memory_offd);
```

Comment:
Is this protected by a printlevel check?

Reply:
Thanks for pointing it out. I should have added print_level guard.

Reply:
Hi @rfalgout, `parcsr_matrix` doesn't have a `print_level`, so I just add `#if defined(HYPRE_DEBUG)` around the `hypre_printf`. Is it OK?

Reply:
Another approach would be to use `hypre_error_w_msg` like here: https://github.com/hypre-space/hypre/blob/ebc634a4e86d4b1aba97f61a8ba0aae701c9799b/src/seq_mv/csr_matrix.c#L177-L178

Reply:
Good point.  Error messages should generally be managed through the error handling functions.

Reply:
Correct! Thanks!